### ISSUE TITLE
build: all.bash takes far too long

### ISSUE THREAD

**Issue by rsc** [golang/go#10571]
all.bash is taking too long. There are many reasons. This is a general tracking bug for work to fix this.

**Comment by josharian** [comment]
One regression: The os/exec race detector test in Go 1.4.2 went from ~0.6s on my machine to ~17.5s with tip. all.bash went from ~2.5m to ~6m, so this accounts for about 15% of the regression.

/cc @dvyukov

**Comment by rsc** [comment]
For the record, although I'm thrilled that the cmd/go tests now run by default, they take 55 seconds on my machine. More to do.

**Comment by bcmills** [comment]
I notice that all.bash uses a tiny fraction of the CPU on my machine, and seems to only execute one process at a time.  Would it be feasible to run more than one test process in parallel?

**Comment by rsc** [comment]
@bcmills, please open a separate issue with details about your machine, Go version, and what `runtime.NumCPU()` returns. Both the build half and the test half of all.bash should consume nearly all the CPUs for most of the time. The start and end of each half tend to get single-threaded due to dependency fanout or fanin (depending on how you look at it), and the initial bootstrap does run single-threaded. But most of all.bash should be very multithreaded.

**Comment by rsc** [comment]
Closing in favor of #26473, which has most of the recent work.
